### PR TITLE
introduce feature

### LINKED ISSUE #108: Highlight search result
In [AssessView](https://github.com/ait-cs-IaaS/Taranis-NG/blob/awake/src/gui/src/views/users/AssessView.vue) if filtered via a search string, each word from the search should be highlighted similar to searching for it via ctrl+f.

It should be possible to disable this behavior (maybe via usersettings?)

Examples on how this could be achived:
https://bitsofco.de/a-one-line-solution-to-highlighting-search-matches/
https://github.com/AlbertLucianto/vue-text-highlight

Note: Maybe something similar could be implemented for the tags

### PR DESCRIPTION
**Pull request by b3n4kh**
Usage: Just input a search text into the "search" field and the "content" will be highlighted.

Filter userinput (search) to prevent ReDoS attacks.

Filter resulting HTML to with https://www.npmjs.com/package/vue-dompurify-html to prevent XSS.

Fixes #108 

image

### CODE REVIEW

**Comment by ewoiset** [comment]
#108 's condition to disable functionality is not implemented or not needed?

**Comment by ewoiset** [comment]
Should it be able to search for whitespace ' ' etc ? a lot of values will be marked if so.

**Comment by ewoiset** [comment]
Search results are only marked if in text. Not e.g. when part of title - this differs from "str+f" behavior

File: src/gui/src/components/assess/card/SummarizedContent.vue [hunk 1]
```diff
@@ -27,15 +31,36 @@ export default {
...
+    const filterStore = useFilterStore()
+    const { newsItemsFilter } = storeToRefs(filterStore)
+    const colorStart = ref(Math.floor(Math.random() * 9))
+
+    const news_item_summary_class = computed(() => {
+      return props.open ? 'news-item-summary-no-clip' : 'news-item-summary'
+    })
+
+    function removeRegexSpecialChars(string) {
+      return string.replace(/[.*+?^${}()<>|[\]\\]/g, '')
```
[the remark this answers is not in the document]
Comment (b3n4kh) [reply src/gui/src/components/assess/card/SummarizedContent.vue]: now checking if content is empty

**Pull request merged**